Implement C wrapper

BOOtak opened the pull request:
Basic C wrapper over WHY_CPP C++ API

BOOtak committed:
Add basic C wrapper for C++ API.

BOOtak committed:
Test context cast magic, add basic drawing C API wrapper.

BOOtak committed:
Implement drawing in the C wrapper.

BOOtak committed:
Hide context from user code.

BOOtak committed:
Add time; import_sprites C API wrappers.

BOOtak committed:
Add lifecycle & text API wrappers.

BOOtak committed:
Add animation support.

BOOtak committed:
Power logging by some magic macros.

BOOtak committed:
Add input support.

BOOtak committed:
Add logging.

senior-sigan commented on whycpp_c/src/log.cpp:
```diff
@@ -0,0 +1,22 @@
+#include <whycpp_c/log.h>
+#include <SDL2/SDL_error.h>
```
this file and entire whycpp_c wrapper should be better not to depend on SDL2 OR we need to configure cmake to include SDL2 headers for this project.
Is there a way to remove this includes? Maybe we need to add some function to the root-whycpp?

BOOtak committed:
Include whycpp headers into whycpp_c project. Move c_api.h from whycpp_c to whycpp as it will be useful there later.

BOOtak committed:
Fix whycpp_c project.

Convert project type from C to CXX;
Remove unneccessary whycpp includes from the whycpp_c project.

BOOtak committed:
Fix various C API definition issues.

BOOtak committed:
Use whycpp logging in whycpp_c, optimize whycpp logging for usage in C code.

BOOtak committed:
Use whycpp buttons in whycpp_c, optimze buttons for usage in C code.

BOOtak commented:
@senior-sigan Apart from the fixes of various issues I found, I've changed C API wrapper to use original WHY_CPP logging
